src/transformers/pipelines/__init__.py

```diff
@@ -859,6 +859,9 @@ def pipeline(
                 'You cannot use both `pipeline(... torch_dtype=..., model_kwargs={"torch_dtype":...})` as those'
                 " arguments might conflict, use only one.)"
             )
+        if torch_dtype is str:
+            if hasattr(torch, torch_dtype):
+                torch_dtype = getattr(torch, torch_dtype)
```

Comment:
* We don't need two separate if indentations
* `x is str` doesn't check for the string type

```suggestion
        if isinstance(torch_dtype, str) and hasattr(torch, torch_dtype):
            torch_dtype = getattr(torch, torch_dtype)
```